linera-views/src/key_value_store_view.rs

```diff
@@ -696,18 +679,16 @@ where
                 results.push(value);
             } else {
                 results.push(false);
-                if !contains_key(&self.deleted_prefixes, &index) {
+                if !self.deletion_set.contains_key(&index) {
                     missed_indices.push(i);
                     let key = self.context.base_tag_index(KeyTag::Index as u8, &index);
                     vector_query.push(key);
                 }
             }
         }
-        if !self.delete_storage_first {
```

Comment:
why was this removed?

Reply:
I think if `delete_storage_first`, then in line 682 we always have `contains_prefix_of(&index)`, so `vector_query` will be empty.